client/src/components/Ticketing/ticketingmanager/Event/components/EventGeneralForm.tsx

```diff
@@ -1,19 +1,55 @@
-import React, {useState, useEffect} from 'react';
-import {Field, Formik} from 'formik';
+import React, {useState, useEffect, useCallback} from 'react';
+import {Field, Formik, useFormikContext} from 'formik';
 import {InputControl} from './InputControl';
 import {eventGeneralSchema} from './event.schemas';
 import {FormSubmitButton} from './FormSubmitButton';
 import {useEvent} from './EventProvider';
 import {EventImage} from '../../../../../utils/imageURLValidation';
 import {getData} from './ShowingUtils';
 import {FormButton} from './FormButton';
+import {useDropzone} from 'react-dropzone';
 import {BackIcon, SaveIcon} from '../../../Icons';
 
 interface EventGeneralFormProps {
   onSubmit: (event, actions) => void;
   onLeaveEdit?: () => void;
 }
 
+const ImageUpload = () => {
+  const {setFieldValue} = useFormikContext();
+
+  const onDrop = useCallback(async (files) => {
+    const formData = new FormData();
+    formData.append('file', files[0]);
+
+    try {
+      const response = await fetch(
+        process.env.REACT_APP_API_2_URL + '/events/image-upload',
+        {
+          method: 'POST',
+          body: formData,
+        },
+      );
+      if (!response.ok) {
+        throw response;
+      }
+      const img = await response.json();
+      setFieldValue('imageurl', img.url);
+    } catch (error) {
+      console.error(error.message);
+    }
+  }, [setFieldValue]);
+
+  const {getRootProps, getInputProps} = useDropzone({onDrop});
+
+  return (
+    <div {...getRootProps()}>
+      <input {...getInputProps()}></input>
+      <p>Drag & Drop File or Click Here to Upload</p>
+    </div>
```

Comment:
For this component, there should be some visual indicator (like a dotted line surrounding the drop zone) that shows that you can drag and drop here. Check out this tailwind-based component for possible styling: https://tailwindcomponents.com/component/dropzone-upload. You may want to make the component thinner than what's on this site though so it visually makes sense.

Let me know if you have any questions!

Reply:
Dotted border has been added to indicate dropzone